packages/extras/src/lib/hooks/useFBO.ts

```diff
@@ -1,67 +1,54 @@
 /* Based on https://github.com/pmndrs/drei/blob/master/src/core/useFBO.tsx under the MIT License */
-import { useThrelte } from '@threlte/core'
-import { onDestroy, onMount } from 'svelte'
-import {
-  type RenderTargetOptions,
-  WebGLRenderTarget,
-  LinearFilter,
-  HalfFloatType,
-  DepthTexture,
-  FloatType
-} from 'three'
 
-interface UseFBOOptions extends RenderTargetOptions {
-  /** Defines the count of MSAA samples. Can only be used with WebGL 2. Default: 0 */
-  samples?: number
-  /** If set, the scene depth will be rendered into buffer.depthTexture. Default: false */
-  depth?: boolean
+import type { RenderTargetOptions } from 'three'
+import { DepthTexture, WebGLRenderTarget } from 'three'
+import { onDestroy } from 'svelte'
+import { isInstanceOf, useThrelte, watch } from '@threlte/core'
+
+export type UseFBOOptions = RenderTargetOptions & {
+  /**
+   * if set, the scene depth will be rendered into buffer.depthTexture
+   */
+  depth?: { width?: number; height?: number } | DepthTexture | boolean
+  /**
+   * if set, the render target size will be set to the corresponding width and height and not use or follow the size of the canvas
+   */
+  dimensions?: { width?: number; height?: number }
 }
 
-// 👇 uncomment when TS version supports function overloads
-// export function useFBO(options?: UseFBOOptions)
-export const useFBO = (
-  /** Width in pixels, or options (will render fullscreen by default) */
-  width?: number | UseFBOOptions,
-  /** Height in pixels */
-  height?: number,
-  /** Options */
-  options?: UseFBOOptions
-): WebGLRenderTarget => {
-  const { dpr, size } = useThrelte()
-
-  const _width = typeof width === 'number' ? width : 1 * (dpr.current ?? 1)
-  const _height = typeof height === 'number' ? height : 1 * (dpr.current ?? 1)
-  const _options = (typeof width === 'number' ? options : (width as UseFBOOptions)) || {}
-  const { samples = 0, depth, ...targetOptions } = _options
-
-  const target = new WebGLRenderTarget(_width, _height, {
-    minFilter: LinearFilter,
-    magFilter: LinearFilter,
-    type: HalfFloatType,
-    ...targetOptions
-  })
-
-  if (depth) {
-    target.depthTexture = new DepthTexture(_width, _height, FloatType)
+export const useFBO = ({
+  depth = false,
+  dimensions,
```

Comment:
Think it makes sense to call this `size` to match the naming convention of the `useThrelte` hook?